MBBSEmu/DOS/ExeRuntime.cs

```diff
@@ -0,0 +1,142 @@
+﻿using MBBSEmu.CPU;
+using MBBSEmu.Disassembler;
+using MBBSEmu.Disassembler.Artifacts;
+using MBBSEmu.DOS.Interrupts;
+using MBBSEmu.DOS.Structs;
+using MBBSEmu.Memory;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using NLog;
+
+namespace MBBSEmu.DOS
+{
+    public class ExeRuntime
+    {
+        public MZFile File;
+        public IMemoryCore Memory;
+        public ICpuCore Cpu;
+        public CpuRegisters Registers;
+        private ILogger _logger;
+
+        /// <summary>
+        ///     Segment which holds the Environment Variables
+        /// </summary>
+        private const ushort ENVIRONMENT_SEGMENT = 0xFFF0;
+
+        /// <summary>
+        ///     Segment which contains the PSP Struct
+        ///
+        ///     Normally +10h beyond the last segment of the EXE
+        /// </summary>
+        private readonly ushort PSP_SEGMENT;
+
+        private readonly List<string> _environmentVariables;
+
+        public ExeRuntime(MZFile file, List<string> environmentVariables, ILogger logger)
```

Comment:
Would environmentVariables be better as a `Dictionary<string, string>` with key/value? I'm assuming right now it's in the form `someVariable=someValue` for each item

Reply:
They're null terminated in memory so I handled them as they're handled by DOS as an array of strings. Since they're written to the Environment Variable sector as `key=value\0` strings, there's not much value in splitting them.

Reply:
Might be useful for later if we grab all the env variables from the C# program and pass them to the running EXE. Either form works, just might require more conversion down the line.